File: API/Entities/Employee.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace API.Entities
{
    public class Employee
    {
        [Key]
        public int EmployeeId { get; set; }

        [Required]
        public string EmployeeName { get; set; }

        [EmailAddress]
        public string EmployeeEmail { get; set; }

        public string EmployeePhone { get; set; }

        public string EmployeeAddress { get; set; }

        public string EmployeeInformation { get; set; }

        // Thông tin chi tiết
        public DateTime? BirthDate { get; set; }

        public string PlaceOfBirth { get; set; }

        public string Gender { get; set; }

        public string MaritalStatus { get; set; }

        public string IdentityNumber { get; set; }

        public DateTime? IdentityIssuedDate { get; set; }

        public string IdentityIssuedPlace { get; set; }

        public string Religion { get; set; }

        public string Ethnicity { get; set; }

        public string Nationality { get; set; }

        public string EducationLevel { get; set; }

        public string Specialization { get; set; }

        // Liên kết phòng ban
        [ForeignKey("Department")]
        public int DepartmentId { get; set; }

        public AppDepartment Department { get; set; }

        // Hợp đồng
        [JsonIgnore]
        public ICollection<Contract> Contracts { get; set; } = new List<Contract>();

        // Lương
        [JsonIgnore]
        public ICollection<Salary> Salaries { get; set; } = new List<Salary>();

        [JsonIgnore]
        public ICollection<TimeKeeping> TimeKeeping { get; set; } = new List<TimeKeeping>();
    }
}

File: client/src/app/_model/employee.ts
export interface Employee {
    EmployeeId: number;
    EmployeeName: string;
    DepartmentId: number;
    EmployeeEmail: string;
    EmployeePhone: string;
    EmployeeAddress: string;
    EmployeeInformation: string;
    BirthDate?: Date;
    PlaceOfBirth?: string;
    Gender?: string;
    MaritalStatus?: string;
    IdentityNumber?: string;
    IdentityIssuedDate?: Date;
    IdentityIssuedPlace?: string;
    Religion?: string;
    Ethnicity?: string;
    Nationality?: string;
    EducationLevel?: string;
    Specialization?: string;
}
